shopfloor/services/cluster_picking.py

```diff
@@ -795,6 +817,11 @@ def scan_destination_pack(self, picking_batch_id, move_line_id, barcode, quantit
                 },
                 qty_done=quantity,
             )
+
+        # Keeping second return value for compatibility
+        # Quantity already checked at _check_line_qty_processible
+        new_line, __ = move_line._split_qty_to_be_done(quantity)
```

Comment:
Could you also extend `test_scan_destination_pack_bin_not_found` and ensure that there is also no new stock.move.line created. 
Because by moving `_split_qty_to_be_done` you fixed also the problem, that there is a stock.move.line created, when a wrong barcode is scanned and you entered a lower qty as `product_uom_qty`.

Reply:
ping @mmequignon

Reply:
Yes sorry, forgot to answer, I'll do this ASAP

Reply:
@mmequignon could you extend the test?

Reply:
I tried to reproduce this behavior with 14.0, no new move line is created after `scan_destination_pack` is called.
```python
        all_lines = self.env["stock.move.line"].search([])
        line = self.one_line_picking.move_line_ids
        response = self.service.dispatch(
            "scan_destination_pack",
            params={
                "picking_batch_id": self.batch.id,
                "move_line_id": line.id,
                # this bin is used for the other picking
                "barcode": "⌿",
                "quantity": line.product_uom_qty,
            },
        )
        new_lines = self.env["stock.move.line"].search([("id", "not in", all_lines.ids)])
        self.assertFalse(new_lines)
```
What am I missing here?

Reply:
The quantity should be less than `line.product_uom_qty`. Only then the split is called.